pkg/network/ovn_kubernetes.go

```diff
@@ -620,7 +621,9 @@ func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootst
 	// On upgrade, we will just remove any existing ipsec deployment without making any
 	// change to them. So during upgrade, we must keep track if IPsec MachineConfigs are
 	// active or not for non Hybrid hosted cluster.
-	isIPsecMachineConfigActive := isIPsecMachineConfigActive(bootstrapResult.Infra)
+	machineConfigPoolPaused := isThereAnyMachineConfigPoolPaused(bootstrapResult.Infra)
+	isIPsecMachineConfigActiveInUnPausedPools := isIPsecMachineConfigActive(bootstrapResult.Infra, true)
```

Comment:
I would move these two variables to the same block where `renderBothIPsecDemonSetsWhenAPoolPausedState` is defined. And then I would elaborate a bit more the comment of that block saying that if there are unpaused pools, we wait until thos poolls have the ipsec machine config active before deploying both daemonsets

Reply:
done